mpi4jax/collective_ops/allreduce.py

```diff
@@ -161,24 +161,115 @@ def mpi_allreduce_value_and_jvp(in_args, tan_args, op, comm):
     res = Allreduce(x, token=token, op=op, comm=comm)
 
     # Identify the correct adjoint
-    op = unpack_hashable(op)
+    op_ = unpack_hashable(op)
 
-    if op == _MPI.SUM:
-        jvp = (x_tan, token_tan)
+    if op_ == _MPI.SUM:
+        jvp = mpi_allreduce_p.bind(x_tan, token, op=op, comm=comm)
```

Comment:
Are you sure about this? 
Because it seems to me in `res = Allreduce(x, token=token, op=op, comm=comm)` you are essentially calling `mpi_allreduce_p.bind(x, token, op comm)` while there you are calling `mpi_allreduce_p.bind(x_tan, token, op=op, comm=comm)`. Shouldn't you be binding to the transpose operation? Or it's smart enough that if the type of inputs are cotangents, he knows it has to evaluate the transpose?

Reply:
I think its correct like this. For linear operators (which the sum is) the jacobian is just the matrix representing the operator, so the jvp is the same as the operator (and the vjp is the same as the transposed operator). I think what is happening when you call the vjp is that jax calculates the transpose of the jvp (which is linear by definition) using our rule.

Reply:
I just was not sure if we should use token_tan instead of token here?